diff --git a/app/controllers/DisconnectController.php b/app/controllers/DisconnectController.php
index abe2006..af72e68 100644
--- a/app/controllers/DisconnectController.php
+++ b/app/controllers/DisconnectController.php
@@ -20,6 +20,7 @@ class DisconnectController extends Base
         // Fresh cookie
         Cookie::renew();
 
-        $this->redirect('login');
+        $main_domain = exec('cat /etc/yunohost/current_host');
+        header('Location: https://' . $main_domain . '/yunohost/sso/?action=logout');
     }
 }
